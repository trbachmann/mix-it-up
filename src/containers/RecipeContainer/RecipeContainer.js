import React, { Component } from 'react';
import { connect } from 'react-redux';
import { RecipeLinkCard } from '../../components/RecipeLinkCard/RecipeLinkCard';
import PropTypes from 'prop-types';
import { UserInstructions } from '../../components/UserInstructions/UserIntructions';

export class RecipeContainer extends Component {
  getRecipesToDisplay = () => {
    const { match } = this.props;
    switch(match.path) {
      case '/my-recipes':
        return this.populateUserRecipes();
      case '/desserts':
      case '/':
        return this.populateRecipesWithoutUsers();
      default:
        return this.populateRecipesWithoutUsers();
    }
  }
  
  populateUserRecipes = () => {
    const { desserts } = this.props;
    const userRecipeCards =  desserts.reduce((acc, recipe) => {
      if (recipe.notes !== '') {
        acc.push(<RecipeLinkCard recipe={recipe} key={recipe.id} />)
      }
      return acc;
    }, []);

    return userRecipeCards.length ? userRecipeCards : (
      <UserInstructions />
    );
  }

  populateRecipesWithoutUsers = () => {
    const { desserts } = this.props;
    return desserts.reduce((acc, recipe) => {
      if (recipe.notes === '') {
        acc.push(<RecipeLinkCard recipe={recipe} key={recipe.id} />)
      }
      return acc;
    }, []);
  }

  render() {
    return(
      <div className='RecipeContainer'>
      {this.getRecipesToDisplay()}
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  desserts: state.desserts
});

export default connect(mapStateToProps)(RecipeContainer);

RecipeContainer.propTypes = {
  desserts: PropTypes.array,
  match: PropTypes.object
}