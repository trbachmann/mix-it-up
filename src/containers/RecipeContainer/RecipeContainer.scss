.RecipeContainer {
  display: grid;
  grid-template-columns: repeat(4, 265px);
  grid-gap: 10px;
  margin: auto auto 20px;
  width: 1090px;
  justify-items: center;
}

@media screen and (max-width: 1100px) {
  .RecipeContainer {
    grid-template-columns: repeat(3, 265px);
    width:815px;
  }
}

@media screen and (max-width: 830px) {
  .RecipeContainer {
    grid-template-columns: repeat(2, 265px);
    width: 540px;
  }
}

@media screen and (max-width: 575px) {
  .RecipeContainer {
    grid-template-columns: 265px;
    width: 275px;
  }
}