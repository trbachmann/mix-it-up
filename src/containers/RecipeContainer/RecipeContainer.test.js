import React from 'react';
import { shallow } from 'enzyme';
import { RecipeContainer, mapStateToProps } from './RecipeContainer';
import * as data from '../../mockData';

describe('RecipeContainer', () => {
  describe('RecipeContainer component', () => {
    let wrapper;
    const mockMatch = { path: '/' };
    
    beforeEach(() => {
      wrapper = shallow(<RecipeContainer desserts={data.mockRecipesWithUserNotes} match={mockMatch}/>)
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should match the snapshot when the path is /my-recipes', () => {
      const mockMatchWithMyRecipes = { path: '/my-recipes' }
      wrapper = shallow(
        <RecipeContainer 
          desserts={data.mockRecipesWithUserNotes} 
          match={mockMatchWithMyRecipes } 
        />
      );
      expect(wrapper).toMatchSnapshot();
    });

    describe('getRecipesToDisplay', () => {
      it.skip('should call populateUserRecipes if /my-recipes', () => {

      });

      it.skip('should call populateRecipesWithOutUsers if /desserts', () => {

      });
    });

    describe('populateUserRecipes', () => {
      it.skip('should return RecipeLinkCards for recipes with notes', () => {

      });
    });

    describe('populateRecipesWithoutUsers', () => {
      it.skip('should returb RecipeLinkCards for recipes without notes', () => {

      });
    });
  });
  
  describe('mapStateToPrpos', () => {
    it('should return an object with desserts', () => {
      const mockState = {
        desserts: data.mockRecipes,
        error: '',
        status: '',
        isLoading: false,
        attribution: data.mockAttribution
      }
      const expected = {
        desserts: data.mockRecipes
      }
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });
});