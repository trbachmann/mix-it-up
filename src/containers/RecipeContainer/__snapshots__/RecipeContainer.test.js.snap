// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeContainer RecipeContainer component should match the snapshot 1`] = `
<div
  className="RecipeContainer"
>
  <RecipeLinkCard
    key="Toasted-marshmallow-cookies-and-cream-milkshake-335047"
    recipe={
      Object {
        "attributes": Object {
          "course": Array [
            "Beverages",
            "Desserts",
          ],
          "cuisine": Array [
            "Kid-Friendly",
          ],
        },
        "attribution": Object {
          "html": "<a href='http://www.yummly.co/recipe/Toasted-marshmallow-cookies-and-cream-milkshake-335047'>Toasted Marshmallow Cookies and Cream Milkshake recipe</a> information powered by <img alt='Yummly' src='https://static.yummly.co/api-logo.png'/>",
          "logo": "https://static.yummly.co/api-logo.png",
          "text": "Toasted Marshmallow Cookies and Cream Milkshake recipes: information powered by Yummly",
          "url": "http://www.yummly.co/recipe/Toasted-marshmallow-cookies-and-cream-milkshake-335047",
        },
        "flavors": Object {
          "Bitter": 0.1667,
          "Meaty": 0.1667,
          "Piquant": 0,
          "Salty": 0.1667,
          "Sour": 0.1667,
          "Sweet": 0.6667,
        },
        "id": "Toasted-marshmallow-cookies-and-cream-milkshake-335047",
        "images": Array [
          Object {
            "hostedLargeUrl": "http://lh4.ggpht.com/eUAHA16-zMjrkoD1qSyD-5GVyckd__2WwY213f-rIi7NfZbCmwfQ53pN9COxZ6q9VZBTbc0weJvJhy8NoVymLQ=s360",
            "hostedMediumUrl": "http://lh4.ggpht.com/eUAHA16-zMjrkoD1qSyD-5GVyckd__2WwY213f-rIi7NfZbCmwfQ53pN9COxZ6q9VZBTbc0weJvJhy8NoVymLQ=s180",
            "hostedSmallUrl": "http://lh4.ggpht.com/eUAHA16-zMjrkoD1qSyD-5GVyckd__2WwY213f-rIi7NfZbCmwfQ53pN9COxZ6q9VZBTbc0weJvJhy8NoVymLQ=s90",
            "imageUrlsBySize": Object {
              "360": "http://lh5.ggpht.com/BtZ6SmVreIVczKceXBa6Chq79PajwSVK047clqtMnSCQ5nlnnfhCEhSl6xzwDoovGG1eVzUnH1NUoHgXazCH=s360-c",
              "90": "http://lh5.ggpht.com/BtZ6SmVreIVczKceXBa6Chq79PajwSVK047clqtMnSCQ5nlnnfhCEhSl6xzwDoovGG1eVzUnH1NUoHgXazCH=s90-c",
            },
          },
        ],
        "ingredientLines": Array [
          "6 Jumbo marshmallows",
          "6 cups good quality vanilla ice cream",
          "1 cup milk",
          "6 Oreo Cookies",
        ],
        "name": "Toasted Marshmallow Cookies and Cream Milkshake",
        "notes": "",
        "numberOfServings": 4,
        "rating": 5,
        "source": Object {
          "sourceDisplayName": "Picky Palate",
          "sourceRecipeUrl": "http://picky-palate.com/2013/02/15/toasted-marshmallow-cookies-and-cream-milkshake/",
          "sourceSiteUrl": "http://picky-palate.com",
        },
        "totalTime": "15 min",
        "totalTimeInSeconds": 900,
        "yield": "4 Servings",
      }
    }
  />
</div>
`;

exports[`RecipeContainer RecipeContainer component should match the snapshot when the path is /my-recipes 1`] = `
<div
  className="RecipeContainer"
>
  <RecipeLinkCard
    key="The-Best-_and-Easy_-Peanut-Butter-Cookies-2665930"
    recipe={
      Object {
        "attributes": Object {
          "course": Array [
            "Afternoon Tea",
            "Desserts",
          ],
          "cuisine": Array [
            "Kid-Friendly",
          ],
        },
        "attribution": Object {
          "html": "<a href='http://www.yummly.co/recipe/The-Best-_and-Easy_-Peanut-Butter-Cookies-2665930'>The Best (and Easy) Peanut Butter Cookies recipe</a> information powered by <img alt='Yummly' src='https://static.yummly.co/api-logo.png'/>",
          "logo": "https://static.yummly.co/api-logo.png",
          "text": "The Best (and Easy) Peanut Butter Cookies recipes: information powered by Yummly",
          "url": "http://www.yummly.co/recipe/The-Best-_and-Easy_-Peanut-Butter-Cookies-2665930",
        },
        "flavors": Object {
          "Bitter": 0.1667,
          "Meaty": 0.8333,
          "Piquant": 0,
          "Salty": 0.3333,
          "Sour": 0,
          "Sweet": 0.8333,
        },
        "id": "The-Best-_and-Easy_-Peanut-Butter-Cookies-2665930",
        "images": Array [
          Object {
            "hostedLargeUrl": "https://lh3.googleusercontent.com/IL4SAQ143siXT-QfTZjeNORYdW5BTxDXLdQw-Ix1-rBMLlTfHKhZZFajd3myiwdIxndI9Kv3mpdzetFJPz7qfQ=s360",
            "hostedMediumUrl": "https://lh3.googleusercontent.com/IL4SAQ143siXT-QfTZjeNORYdW5BTxDXLdQw-Ix1-rBMLlTfHKhZZFajd3myiwdIxndI9Kv3mpdzetFJPz7qfQ=s180",
            "hostedSmallUrl": "https://lh3.googleusercontent.com/IL4SAQ143siXT-QfTZjeNORYdW5BTxDXLdQw-Ix1-rBMLlTfHKhZZFajd3myiwdIxndI9Kv3mpdzetFJPz7qfQ=s90",
            "imageUrlsBySize": Object {
              "360": "https://lh3.googleusercontent.com/JwhZ5F6bm4I67nr2WFSuv_anMW33tuBcWk1n7zzu6zIaAJraUPoogaJexrDUKsIQ2nrEWwtmxlFGSr533anG=s360-c",
              "90": "https://lh3.googleusercontent.com/JwhZ5F6bm4I67nr2WFSuv_anMW33tuBcWk1n7zzu6zIaAJraUPoogaJexrDUKsIQ2nrEWwtmxlFGSr533anG=s90-c",
            },
          },
        ],
        "ingredientLines": Array [
          "1/4 cup creamy peanut butter (can use natural one too)",
          "2 tablespoons unsalted butter , room temperature (not melted - just let the butter from refrigerator sit on the counter for about an hour or so)",
          "1/2 cup brown sugar",
          "1 egg",
          "1/2 teaspoon pure vanilla extract",
          "3/4 cup all-purpose flour , sifted or aerated (see important note below!)",
          "1/2 teaspoon baking powder",
          "2 tablespoons granulated sugar for coating",
        ],
        "name": "The Best (and Easy) Peanut Butter Cookies",
        "notes": "Increase peanut butter by 1/2 cup and increase vanilla 2 teaspoons",
        "numberOfServings": 2,
        "rating": 4,
        "source": Object {
          "sourceDisplayName": "Julia's Album",
          "sourceRecipeUrl": "https://juliasalbum.com/easy-peanut-butter-cookies/",
          "sourceSiteUrl": "juliasalbum.com",
        },
        "totalTime": "40 min",
        "totalTimeInSeconds": 2400,
        "yield": "serves two",
      }
    }
  />
</div>
`;
